### PR TITLE
feat(Dataset): add initial background queue, task, work and trigger t…

### PR DESCRIPTION
**Pull request by nazarfil**
Creates initial background queue set-up

### CODE REVIEW

File: docker-compose.yaml [hunk 1]
```diff
@@ -89,6 +89,15 @@ services:
     depends_on:
       - db
 
+  dataset_file_metadata:
```
Comment (qgerome) [inline docker-compose.yaml:92]: `datasets_worker` may be better since we are going to do more things than just metadata extraction

**Review by qgerome — CHANGES_REQUESTED** [review]

File: docker-compose.yaml (hunk 1 above)
Comment (nazarfil) [reply docker-compose.yaml:92]: Yes, it is a better name

File: hexa/datasets/schema/mutations.py [hunk 2]
```diff
@@ -252,11 +253,19 @@ def resolve_create_version_file(_, info, **kwargs):
                     uri=version.get_full_uri(mutation_input["uri"]),
                     content_type=mutation_input["contentType"],
                 )
-                return {
-                    "success": True,
-                    "errors": [],
-                    "file": file,
+            dataset_file_metadata_queue.enqueue(
```
Comment (qgerome) [inline hexa/datasets/schema/mutations.py:256]: I think we should add a `load_metadata` on the DatasetVersionFile and call that on the save() method if there are no metadata attached.

**Review by qgerome — CHANGES_REQUESTED** [review]

File: docker-compose.yaml [hunk 3]
```diff
@@ -89,6 +89,15 @@ services:
     depends_on:
       - db
 
+  dataset_worker:
+    <<: *common
+    command: "manage dataset_worker"
+    restart: unless-stopped
+    profiles:
```
[the remark this answers is not in the document]
Comment (nazarfil) [reply docker-compose.yaml:96]: Added it to  the task PR, will move it here

File: hexa/datasets/schema/mutations.py (hunk 2 above)
Comment (nazarfil) [reply hexa/datasets/schema/mutations.py:256]: So on creation of DatasetVersionFile you suggest to change the flow to trigger generation of metadata file on save:
- DatasetVersionFile.save() -> load_metadata() -> enqueue(task)
Where load_metadata will enqueue the sample generation task, which does all the job to download, link with FK to DatasetVersionFile and save in db the sample

**Comment by qgerome** [comment]
Yes

**Review by qgerome — APPROVED** [review]

**Pull request merged**